hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -318,6 +323,7 @@ protected void completeCompaction(HoodieCommitMetadata metadata, JavaRDD<WriteSt
   @Override
   protected JavaRDD<WriteStatus> compact(String compactionInstantTime, boolean shouldComplete) {
     HoodieSparkTable<T> table = HoodieSparkTable.create(config, context);
+    table.getHoodieView().sync();
     preWrite(compactionInstantTime, WriteOperationType.COMPACT, table.getMetaClient());
```

Comment:
Is there any possibility that we only sync the view when metadata table is enabled ?

Reply:
seems like something we should do always anyway. do you see any issues around how this may not work well for Flink?

Reply:
@danny0405 : I am in need of this for [metadata sync patch](https://github.com/apache/hudi/pull/3590). If you don't mind, I will land this patch now. But we can discuss and get to a consensus. If need be, I can put up a follow patch to make it conditional.